config/gatewaycrd/v1.2.0.yaml

```diff
@@ -0,0 +1,10345 @@
+# Copyright 2024 The Kubernetes Authors.
```

Comment:
maybe this should be in the config/crd folder? "config/crd/gateway"

Reply:
where is this used? I don't see this used anywhere in this code

Reply:
it's required to create our mock client for tests - see https://github.com/Azure/aks-app-routing-operator/pull/306/files#diff-63440271bd8dc852878ea83b0c4d954a51f8e05348014b61c2f1bfbb16bc99a4R131